### PR TITLE
2-jws-MultilayerBU

### LINKED ISSUE #2: Multilayer Graphs
Create:
- [ ] _MultilayerWD.gen.m (Blanca)
- [x] #12 (Hang)
- [x] _MultilayerBD.gen.m (Blanca)
- [x] #19

Create:
- [x] _OrdMlWD.gen.m (Hang)
- [x] _OrdMlWU.gen.m (Hang)
- [ ] _OrdMlBD.gen.m (Blanca)
- [ ] _OrdMlBU.gen.m (jws)

Create
- [x] _OrdMxBUD.gen.m (Hang)
- [x] _OrdMxBUT.gen.m (Blanca)
- [x] #11 (jws)
- [x] _MultilayerBUT.gen.m (Blanca)
- [ ] _OrdMlBUD.gen.m (jws)
- [ ] _OrdMlBUT.gen.m (jws)

### LINKED ISSUE #19: _MultilayerBU.gen.m (jws)

### PR DESCRIPTION
**Pull request by JiaweiSun0512**
1._MultilayerBU.gen.m
2. measures

### CODE REVIEW

**Review by hanzh346 — APPROVED** [review]

**Review by hanzh346 — APPROVED** [review]

File: braph2genesis/graphs/_MultilayerBU.gen.m [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+B = g.get('B'); %#ok<PROPLC>
+L = length(B); %#ok<PROPLC> % number of layers
+for i = 1:1:L
+    M = symmetrize(B{i, i}, 'SymmetrizeRule', g.get('SYMMETRIZE_RULE')); %#ok<PROPLC> % enforces symmetry of adjacency matrix
+    M = dediagonalize(M); % removes self-connections by removing diagonal from adjacency matrix, equivalent to dediagonalize(M, 'DediagonalizeRule', 0)
+    M = semipositivize(M, 'SemipositivizeRule', g.get('SEMIPOSITIVIZE_RULE')); % removes negative weights
+    M = binarize(M); % enforces binary adjacency matrix, equivalent to binarize(M, 'threshold', 0, 'bins', [-1:.001:1])
+    B(i, i) = {M};
+end
+A = B;
```
Comment (iok-ui) [inline braph2genesis/graphs/_MultilayerBU.gen.m:104]: Is there particular reason to have this line instread of "value = B;"?

File: braph2genesis/graphs/_MultilayerBU.gen.m [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+    B13'                         B23'                           B3
+    };
+g = MultilayerBU('B', B);
+g.get('A_CHECK')
+A1 = symmetrize(binarize(semipositivize(dediagonalize(B1))));
+A2 = symmetrize(binarize(semipositivize(dediagonalize(B2))));
+A3 = symmetrize(binarize(semipositivize(dediagonalize(B3))));
+B{1,1} = A1;
+B{2,2} = A2;
+B{3,3} = A3;
```
Comment (iok-ui) [inline braph2genesis/graphs/_MultilayerBU.gen.m]: leave a space after comma, e.g. 
B12 = rand(size(B1, 1), size(B2, 2));

And spread it across the whole script

**Review by iok-ui — CHANGES_REQUESTED** [review]

**Review by egolol — APPROVED** [review]

File: braph2genesis/graphs/_MultilayerBU.gen.m [hunk 3]
```diff
@@ -0,0 +1,433 @@
+% ¡header!
+MultilayerBU < Graph (g, multilayer binary undirected graph) is a multilayer binary undirected graph.
+
+%%% ¡description!
+In a multilayer binary undirected (BU) graph, layers have same or different number 
+ of nodes with within-layer undirected edges either 0 (absence of connection) 
+ or 1 (existence of connection).
+The connectivity matrices are symmetric.
+There are connections between layers connecting the corresponding nodes.
```
Comment (annacanal) [inline braph2genesis/graphs/_MultilayerBU.gen.m]: In a multilayer binary undirected (BU) graph, layers could have different number 
 of nodes with within-layer undirected edges. Edges can be either 0 (absence of connection)  or 1 (existence of connection).
The connectivity matrices are symmetric (within layer).
All node connections are allowed between layers.

File: braph2genesis/graphs/_MultilayerBU.gen.m [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+%%% ¡prop!
+
+NAME (constant, string) is the name of the multilayer binary undirected graph.
+%%%% ¡default!
+'MultilayerBU'
+
+%%% ¡prop!
+DESCRIPTION (constant, string) is the description of the multilayer binary undirected graph.
+%%%% ¡default!
+'In a multilayer binary undirected (BU) graph, layers have same or different nodes with within-layer undirected edges either 0 (absence of connection) or 1 (existence of connection). The connectivity matrices are symmetric. There are connections between layers connecting the corresponding nodes.'
```
Comment (annacanal) [inline braph2genesis/graphs/_MultilayerBU.gen.m]: In a multilayer binary undirected (BU) graph, layers could have different number of nodes with within-layer undirected edges. Edges can be either 0 (absence of connection) or 1 (existence of connection). The connectivity matrices are symmetric (within layer). All node connections are allowed between layers.

File: braph2genesis/graphs/_MultilayerBU.gen.m [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+%%%% ¡calculate!
+if isempty(varargin)
+    layernumber = 1;
+else
+    layernumber = varargin{1};
+end
+value = Graph.NONNEGATIVE * ones(layernumber);
+
+%%% ¡prop!
+A (result, cell) is the cell containing the multilayer binary adjacency matrices of the multilayer binary undirected graph.
```
Comment (annacanal) [inline braph2genesis/graphs/_MultilayerBU.gen.m]: A (result, cell) is the cell containing the binary supra-adjacency matrix of the multilayer binary undirected graph.

**Review by annacanal — CHANGES_REQUESTED** [review]
All descriptions were wrong, please update them as soon as possible.

File: braph2genesis/graphs/_MultilayerBU.gen.m [hunk 6]
```diff
@@ -90,7 +90,7 @@ In a multilayer binary undirected (BU) graph, layers could have different number
 value = Graph.NONNEGATIVE * ones(layernumber);
 
 %%% ¡prop!
-A (result, cell) is the cell containing the multilayer binary adjacency matrices of the multilayer binary undirected graph.
+A (result, cell) is the cell containing the multilayer binary supra-adjacency matrices of the multilayer binary undirected graph.
```
Comment (annacanal) [inline braph2genesis/graphs/_MultilayerBU.gen.m]: A (result, cell) is the cell containing the binary supra-adjacency **matrix** of the multilayer binary undirected graph.

**Review by annacanal — CHANGES_REQUESTED** [review]

File: braph2genesis/graphs/_MultilayerBU.gen.m (hunk 6 above)
Comment (annacanal) [reply braph2genesis/graphs/_MultilayerBU.gen.m]: this is still not changed

**Review by annacanal — APPROVED** [review]

**Review by bzufiriaki — APPROVED** [review]

**Pull request merged**